lib/kamal/commander.rb

```diff
@@ -28,11 +28,34 @@ def specific_primary!
   end
 
   def specific_roles=(role_names)
-    @specific_roles = config.roles.select { |r| role_names.include?(r.name) } if role_names.present?
+    return unless role_names.present?
+
+    @specific_roles = []
+
+    Array(role_names).select do |sp|
+      @specific_roles += config.roles.select do |r|
+        # Only allow * for a wildcard
+        pattern = Regexp.escape(sp).gsub('\*', '.*')
```

Comment:
Could you extract this to a method in `Kamal::Utils`?

Reply:
Done! Looks a bit cleaner.